app/src/addr.c

```diff
@@ -36,24 +51,63 @@ zxerr_t addr_getItem(int8_t displayIdx,
                      char *outVal, uint16_t outValLen,
                      uint8_t pageIdx, uint8_t *pageCount) {
     zemu_log_stack("addr_getItem");
-    switch (displayIdx) {
-        case 0:
+
+    if (is_pubkey(show_address) && displayIdx--==0) {
             snprintf(outKey, outKeyLen, "Pub Key");
-            // +2 is to skip 0x04 prefix that indicates uncompresed key 
-            pageString(outVal, outValLen, (char *) (G_io_apdu_buffer + VIEW_ADDRESS_OFFSET_SECP256K1 + 2), pageIdx, pageCount);
+            // +1 is to skip 0x04 prefix that indicates uncompresed key 
+            pageHexString(outVal, outValLen, pubkey_to_display+1, sizeof(pubkey_to_display)-1, pageIdx, pageCount);
             return zxerr_ok;
-        case 1: {
-            if (!app_mode_expert()) {
-                return zxerr_no_data;
+    }
+
+    if (displayIdx--==0) {
+            switch(show_address) {
+                case show_address_error:
+                    snprintf(outKey, outKeyLen, "Error");
+                    pageString(outVal, outValLen, "deriving public key.", pageIdx, pageCount);
+                    return zxerr_ok;
+                case show_address_no_pubkey:
+                    snprintf(outKey, outKeyLen, "Account data");
+                    pageString(outVal, outValLen, "not saved on the device.", pageIdx, pageCount);
+                    return zxerr_ok;
+                case show_address_empty_slot:
+                    snprintf(outKey, outKeyLen, "Address:");
+                    pageString(outVal, outValLen, "Not saved on the device.", pageIdx, pageCount);
+                    return zxerr_ok;
+                case show_address_hdpaths_not_equal:
+                    snprintf(outKey, outKeyLen, "Address:");
+                    pageString(outVal, outValLen, "Other path is saved on the device.", pageIdx, pageCount);
+                    return zxerr_ok;
+                case show_address_yes:
+                    snprintf(outKey, outKeyLen, "Address:");
+                    pageHexString(outVal, outValLen, address_to_display.data, sizeof(address_to_display.data), pageIdx, pageCount);
+                    return zxerr_ok;
+                default:
+                    return zxerr_no_data;
             }
+    }
+
+    if (show_address == show_address_yes && displayIdx--==0) {
+        snprintf(outKey, outKeyLen, "Verify if this");
+        snprintf(outVal, outValLen, " public key was   added to");
+        return zxerr_ok;
+    }
+
+    if (show_address == show_address_yes && displayIdx--==0) {
+        array_to_hexstr(outKey, outKeyLen, address_to_display.data, sizeof(address_to_display.data)); 
+#if defined(TARGET_NANOX)
+        snprintf(outVal, outValLen, " using any Flow blockchain explorer.");
+#else
+        snprintf(outVal, outValLen, " using any Flow  blockch. explorer.");
```

Comment:
Why the abbreviation here?

Reply:
So it fits on the screen. Even this way, the last dot is not displayed (maybe I should remove it, but snprintf removes it anyway) This is the best solution I was able to come with. If you have some other idea or you prefer splitting the text into another screen, let me know.

Reply:
Let's go with it as it for now, please.